firmware/lm32/vga_in.c

```diff
@@ -0,0 +1,197 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <irq.h>
+#include <uart.h>
+#include <time.h>
+#include <system.h>
+#include <generated/csr.h>
+#include <generated/mem.h>
+#include <hw/flags.h>
+
+#include "vga_in.h"
+
+int vga_in_debug;
+int vga_in_fb_index;
+
+#define FRAMEBUFFER_COUNT 4
+#define FRAMEBUFFER_MASK (FRAMEBUFFER_COUNT - 1)
+
+#define BYTES_PER_PIXEL 2 // should be changed for 32-bits per pixel
+
+#define VGA_IN_FRAMEBUFFERS_BASE 0x00000000
+#define VGA_IN_FRAMEBUFFERS_SIZE 1024*768*BYTES_PER_PIXEL  //intially was 1280*720*2
+
+
+unsigned int vga_in_framebuffer_base(char n) {
+    return VGA_IN_FRAMEBUFFERS_BASE + n*VGA_IN_FRAMEBUFFERS_SIZE;
+}
+
+static int vga_in_fb_slot_indexes[2];
+static int vga_in_next_fb_index;
+static int vga_in_hres, vga_in_vres;
+
+void vga_in_isr(void)
```

Comment:
Is this code identical to the HDMI version? Or does it have some customisations?